### PR TITLE
chore: 🔧 Upgrade to pnpm 9

### LINKED ISSUE #440: chore: 🔧 Upgrade environment to use pnpm@9
# Description

A new [major pnpm version was released](https://github.com/pnpm/pnpm/releases/tag/v9.0.0) some time ago. It adds better support for resolving merge conflicts in its lockfile. However, the new version of the lockfile is incompatible with version 8, so this change would need an upgrade to all developer machines and gitlab-ci runners to use the new version.

It now also provides support for [corepack](https://nodejs.org/api/corepack.html) via the package.json `packageManager` field. This should make it easier to force some version of pnpm in the future.

# Rationale

Staying up to date with our build tools is crucial so we are able to benefit from updates.

### PR DESCRIPTION
**Pull request by kirchsuSICKAG**
# Pull Request

## 📖 Description
This PR upgrades pnpm to the latest version (9.14.2)
With this PR a `packageManager` field is added to the `package.json`, that makes sure, that always the same pnpm version is used.
Because of that the pnpm version in the github actions could also be removed, because it will also always that the version of the `packageManager` property

### 🎫 Issues
Closes #440 

## 👩‍💻 Reviewer Notes

If not already enabled, it might be necessary to enable corepack. If this is the case have a look at https://pnpm.io/installation#using-corepack

## 📑 Test Plan

Have a look at everything, if nothing is broken because of the change.

## ✅ DoD

- [x] I have tested my changes manually.
- [ ] I have added automatic tests for my changes (unit- and visual regression tests).
- [ ] I have updated the project documentation to reflect my changes (e.g. CHANGELOG, Storybook, ...).
- [ ] I have added documentation to the DaVinci migration guide (if need be)
- [ ] I have made sure to follow the projects coding and contribution guides.
- [ ] I have made sure that the bundle size has changed appropriately.
- [ ] I have validated that there are no accessibility errors.
- [ ] I have used design tokens instead of fix css values

### CODE REVIEW

**Review by schilchSICKAG — CHANGES_REQUESTED** [review]
This is just minor nitpicking and I would be fine without it, but something in my inner OCD says we should make sure the minimal required version for corepack should be node@20, too :). Otherwise, I am fine with this PR, nice to have this finally be done

File: package.json [hunk 1]
```diff
@@ -11,7 +11,7 @@
   "license": "SEE LICENSE IN LICENSE",
   "engines": {
     "node": ">=18.17.0",
```
[the remark this answers is not in the document]
Comment (kirchsuSICKAG) [reply package.json]: Done

**Review by schilchSICKAG — COMMENTED** [review]
Hmm, one last (litte, I promise ^^) thing: Could you add the information we are using corepack to the core `README.md` of the project (ca line#35)? This would help if someone wants to work with our repo and does not have corepack installed. Something in the line of:

> SDS [SIC, should be Synergy?] uses a monorepo setup, using [node.js](https://nodejs.org) as a runtime and [pnpm](https://pnpm.io) for dependency management and build scripts. We also use [corepack](https://pnpm.io/installation#using-corepack) to handle the current required versions of node and pnpm. If not already installed, install and configure these tools.

**Review by schilchSICKAG — APPROVED** [review]
Thanks, very nice, lets give it a go

**Pull request merged**